File: projects/cairo/read_cod_tables.R
library(openxlsx)
library(dplyr)

tables.path <- paste0(dirname(rstudioapi::getActiveDocumentContext()$path), "/db_data/tables")
tables <- list.files(path = tables.path)

ltables <- list()
for(i in tables){
  a.table <- paste0(tables.path, "/", i)
  ltables <- append(ltables, a.table)
}

ltables

# read tables
condition <- read.xlsx(ltables[[1]])
dating <- read.xlsx(ltables[[2]])
features <- read.xlsx(ltables[[3]])
featuresnumbers <- read.xlsx(ltables[[4]])
photos <- read.xlsx(ltables[[5]])
records <- read.xlsx(ltables[[6]])
# records1 <- read.xlsx(ltables[[7]])
smallpics <- read.xlsx(ltables[[8]])

### Reference data
write.table(condition, 
            "C:/Rprojects/eamena-arches-dev/projects/cairo/reference_data/condition.csv", 
            sep = ",", row.names = F)


## HP
records <- merge(records, photos, by = "ID", all.x = T)
df <- records
df <- merge(df, condition, by = "conditionID", all.x = T)
df <- df %>%
  select(where(function(x) any(!is.na(x))))
df <- merge(df, smallpics, by.x = "ID", by.y = "UnitNumber", all.x = T)
df <- df %>%
  select(where(function(x) any(!is.na(x))))
df <- merge(df, dating, by.x = "datetypenumber", by.y = "TypeNumber", all.x = T)
df <- df %>%
  select(where(function(x) any(!is.na(x))))


df[nrow(df) + 1,] <- NA
write.table(df, 
            "C:/Rprojects/eamena-arches-dev/projects/cairo/business_data/hp.csv",
            sep = ",", row.names = F)

# View(df)
# colnames(df)

# BC
# Features as Built Components
df <- features
# df <- merge(df, features, by = "ID", all.x = T)
# df <- df %>%
#   select(where(function(x) any(!is.na(x))))
df <- merge(df, featuresnumbers, by.x = "featureID1", by.y = "featureID", all.x = T)
df <- df %>%
  select(where(function(x) any(!is.na(x))))

df[nrow(df)+1,] <- NA
write.table(df, "C:/Rprojects/eamena-arches-dev/projects/cairo/business_data/bc.csv",
            sep = ",", row.names = F)

# View(df)
# colnames(df)

File: projects/cod/code/read_cod_tables.py
# %%

import pandas as pd
import os


# base_path = os.path.dirname(__file__)
# tables_path = os.path.join(base_path, "db_data", "tables")
# tables = os.listdir(tables_path)
# ltables = [os.path.join(tables_path, table) for table in tables]



# %%

base_path = os.path.dirname(__file__)
tables_path = os.path.join(base_path, "db_data", "tables")
tables = os.listdir(tables_path)
ltables = [os.path.join(tables_path, table) for table in tables]

# Read the Excel tables
condition = pd.read_excel(ltables[0])
dating = pd.read_excel(ltables[1])
features = pd.read_excel(ltables[2])
featuresnumbers = pd.read_excel(ltables[3])
photos = pd.read_excel(ltables[4])
records = pd.read_excel(ltables[5])
# records1 = pd.read_excel(ltables[6])
smallpics = pd.read_excel(ltables[7])

# Save the 'condition' dataframe to CSV
condition.to_csv("C:/Rprojects/eamena-arches-dev/projects/cairo/reference_data/condition.csv", index=False)

# Merge tables and perform data selection in HP
records = records.merge(photos, on="ID", how="left")
df = records.merge(condition, on="conditionID", how="left")
df = df.dropna(how='all', axis=1)
df = df.merge(smallpics, left_on="ID", right_on="UnitNumber", how="left")
df = df.dropna(how='all', axis=1)
df = df.merge(dating, left_on="datetypenumber", right_on="TypeNumber", how="left")
df = df.dropna(how='all', axis=1)

df.loc[len(df)] = pd.NA
df.to_csv("C:/Rprojects/eamena-arches-dev/projects/cairo/business_data/hp.csv", index=False)

# Merge features for BC (Built Components)
df = features.merge(featuresnumbers, left_on="featureID1", right_on="featureID", how="left")
df = df.dropna(how='all', axis=1)
df.loc[len(df)] = pd.NA
df.to_csv("C:/Rprojects/eamena-arches-dev/projects/cairo/business_data/bc.csv", index=False)


# %%

import os

# Example file path
file_path = '/path/to/your/file/example.xlsx'

# Extract the base name (filename with extension)
file_name_with_extension = os.path.basename(file_path)

# Remove the extension
file_name_without_extension = os.path.splitext(file_name_with_extension)[0]

print("File name with extension:", file_name_with_extension)
print("File name without extension:", file_name_without_extension)

# %%
